### DISCUSSION TITLE
Same size despite different sizes

### DISCUSSION THREAD

**Discussion by Individuum92** [ImageMagick/ImageMagick#7710] [General]
Hello everyone,

I need a small batch script for the Windows11 environment, which automatically organizes portrait / landscape images.
All images are located in folder X. The respective orientation (landscape and portrait) always have the same size.

So:
If the picture has the dimensions A x B, then it is in the portrait folder.
If the image has the dimensions C x D, then the folder is landscape.

I have already worked with magick identify. But here, at least to my understanding, the correct values are not output, which means that my automation does not work:

Exemplary 2 images:
![1](https://github.com/user-attachments/assets/eecf8d0e-56e7-43a4-a225-da2599ecf6a2)

And the output of the tool when reading the images:
![2](https://github.com/user-attachments/assets/6ac04c75-1c4e-4ac6-8d61-ba2ef1b383dc)

Why are the sizes of the two images the same?

**Comment by snibgo** [comment]
I expect the image in KRA7201.JPG is 6192x4128 pixels, but with metadata that says the image should be rotated 90 degrees clockwise or anticlockwise. You can do this with "-auto-orient".
```
magick KRA7201.JPG -auto-orient info:
```
Does that give the result you expect?

**Comment by Individuum92** [comment]
Unfortunately, it doesnt work for me:

`C:\Users\kraus\Desktop\IN>magick identify KRA07296.JPG -auto-orient
KRA07296.JPG JPEG 6192x4128 6192x4128+0+0 8-bit sRGB 15.6827MiB 0.001u 0:00.001

C:\Users\kraus\Desktop\IN>magick identify KRA07201.JPG -auto-orient
KRA07201.JPG JPEG 6192x4128 6192x4128+0+0 8-bit sRGB 15.8374MiB 0.001u 0:00.001`

**Reply by Individuum92** [reply]
Ah, sorry, Did not read it carefully enough ...

Yes. Your commands output looks much better:
KRA07201.JPG JPEG 4128x6192 4128x6192+0+0 8-bit sRGB 15.8374MiB 0.010u 0:00.009

Okay. I'll try to implement this in my script to automate it.

**Comment by snibgo** [comment]
What about the command I showed you?

**Comment by snibgo** [comment]
Okay, so the problem was orientation. We can then use `-format` like this ...
```
magick wonderfit.JPG -auto-orient -format "%[fx:w>h?1:0]" info:
```
... to get "1" when the width is greater than the height, otherwise "0". You can put that in a `for` loop to put the 1 or 0 in an environment variable, then an `if` test executes a move to a chosen directory.
